src/lib.rs

```diff
@@ -113,10 +126,26 @@ macro_rules! nonmax {
                 assert_eq!(size_of::<$primitive>(), size_of::<$nonmax>());
                 assert_eq!(size_of::<$nonmax>(), size_of::<Option<$nonmax>>());
             }
+
+            #[test]
+            fn convert() {
+                use std::convert::TryFrom;
+                let zero = $nonmax::try_from(0 as $primitive).unwrap();
+                let zero = $primitive::from(zero);
+                assert_eq!(zero, 0);
+
+                $nonmax::try_from($primitive::max_value()).unwrap_err();
+            }
         }
     };
 }
 
+fn try_from_int_error_overflow() -> std::num::TryFromIntError {
+    // Why yes, we *do* need to resort to this ugliness just to create a TryFromIntError!
+    use std::convert::TryFrom;
+    u8::try_from(999u32).unwrap_err() // infalliable
+}
```

Comment:
This function feels incredibly sketchy! I appreciate that it gives us the same error type as `NonZero*`, which makes us pretty much interchangeable, but... at what cost?

I think we should do one of two things:
1. File a Rust issue to expose a constructor for `TryFromIntError`. You can construct them... sorta, anyways. After that, we should link to that issue from this code so that there's a discussion for context.
2. Create a new error type. I _suspect_ this is what would be recommended to us by people who work on std. It means that nonmax can't be substituted as-is in nonzero code, but that might be okay.

Reply:
> at what cost?

It's a little ugly, and in debug builds there's a bit of cruft... but that's about it, I think.  It optimizes away entirely in optimized builds: https://rust.godbolt.org/z/fnP63j
```
example::try_from_int_error_overflow:
        ret
```
In theory std could add a payload to TryFromIntError that includes the original value that overflowed for better error reporting, which would be wrong with this code, but that sounds kind of expensive for such a core std operation.  Even ParseIntError only exposes [IntErrorKind](https://doc.rust-lang.org/std/num/enum.IntErrorKind.html), and that's not yet stabilized, and choosing 999 as I have would give the "correct" result of IntErrorKind::Overflow.

> 1. File a Rust issue to expose a constructor for TryFromIntError.

It's niche enough that I'd hesitate to increase std's API surface just to get rid of a single ugly fn here.  If anything, I'd rather try to upstream this crate into `std::num::*`! 😄

> 2. Create a new error type.

I can do this if desired.